specutils/spectra/spectrum1d.py

```diff
@@ -95,13 +101,29 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
         # parse a given wcs. This is put into a GWCS object to
         # then be used behind-the-scenes for all specutils operations.
         if spectral_axis is not None:
-            # Ensure that the spectral axis is an astropy quantity
+            # Ensure that the spectral axis is an astropy Quantity
             if not isinstance(spectral_axis, u.Quantity):
-                raise ValueError("Spectral axis must be a `Quantity` object.")
+                raise ValueError("Spectral axis must be a `Quantity` or `SpectralCoord` object.")
+
+            # If spectral axis is provided as an astropy Quantity, convert it
+            # to a specutils SpectralCoord object.
+            if not isinstance(spectral_axis, SpectralCoord):
```

Comment:
To be more inline with the comment and because it's more explicitly what we want to check (i.e. that it's a quantity).

```suggestion
            if isinstance(spectral_axis, u.Quantity):
```

Reply:
Not implementing since, as mentioned above, checking on u.Quantity won't discriminate between Quantity and SpectralCoord